x-pack/filebeat/input/httpjson/input.go

```diff
@@ -175,8 +176,14 @@ func runWithMetrics(ctx v2.Context, cfg config, pub inputcursor.Publisher, crsr
 }
 
 func run(ctx v2.Context, cfg config, pub inputcursor.Publisher, crsr *inputcursor.Cursor, reg *monitoring.Registry) error {
-	log := ctx.Logger.With("input_url", cfg.Request.URL)
+	stat := ctx.StatusReporter
+	if stat == nil {
+		stat = noopReporter{}
+	}
+	stat.UpdateStatus(status.Starting, "")
+	stat.UpdateStatus(status.Configuring, "")
```

Comment:
Does two different statuses here make sense since they seem to basically happen instantaneously of each other ?
Can we not combine them ?

Reply:
They get written into the logs as well. I was wondering about this, but event with arbitrarily distant status changes, we cannot guarantee that a user will see a status notification in the UI, so this is qualitatively no different to any other sequence of state changes.

I personally think that the allowable changes in the state machine that the API represents should be documented in the interface so that this becomes less ambiguous. cc @pkoutsovasilis

Reply:
@pkoutsovasilis No, I was thinking about the documentation. The state diagram (or some representation of it) would be helpful [somewhere here](https://github.com/elastic/beats/blob/6b6941eed4967419043c6fe9810f9f9ea2f042cc/libbeat/management/status/status.go#L18-L49), I guess with an indication that the status is already at Starting when the reporter is handed to the input.